api/wikis/serializers.py

```diff
@@ -99,6 +108,40 @@ class NodeWikiSerializer(WikiSerializer):
         related_view_kwargs={'wiki_id': '<_id>'},
     )
 
+    def update(self, instance, validated_data):
+        auth = Auth(self.context['request'].user)
+        new_page_name = validated_data.get('page_name')
+        validated_data.pop('content', None)
+
+        try:
+            instance = instance.node.rename_node_wiki(instance.page_name, new_page_name, auth)
+        except (
+            NameInvalidError, NameMaximumLengthError,
+            PageCannotRenameError
```

Comment:
Rather than putting handling all these, you could do

```
except PageConflict as err:
    raise Conflict(...)
except WikiError...:
   raise ValidationError(...)
```